SDWebImageAVIFCoder/Classes/SDImageAVIFCoder.m

```diff
@@ -13,41 +14,610 @@
 #import "avif/avif.h"
 #endif
 
-// Convert 8/10/12bit AVIF image into RGBA8888
-static void ConvertAvifImagePlanarToRGB(avifImage * avif, uint8_t * outPixels) {
-    avifBool usesU16 = avifImageUsesU16(avif);
+static void SetupConversionInfo(avifImage * avif,
+                                avifReformatState* state,
+                                vImage_YpCbCrToARGBMatrix* matrix,
+                                vImage_YpCbCrPixelRange* pixelRange) {
+    avifPrepareReformatState(avif, state);
+
+    // Setup Matrix
+    matrix->Yp = 1.0f;
+    matrix->Cr_R = 2.0f * (1.0f - state->kr);
+    matrix->Cb_B = 2.0f * (1.0f - state->kb);
+    matrix->Cb_G = -2.0f * (1.0f - state->kr) * state->kr / state->kg;
+    matrix->Cr_G = -2.0f * (1.0f - state->kb) * state->kb / state->kg;
+    
+    // Setup Pixel Range
+    switch (avif->depth) {
+        case 8:
+            if (avif->yuvRange == AVIF_RANGE_LIMITED) {
```

Comment:
Does there are something built-in pixel range instead of declaration them hardcodely ?

Reply:
It looks there are no emus:
https://github.com/AOMediaCodec/libavif/blob/master/include/avif/avif.h#L285

libavif developers uses hard-coded values directly:
https://github.com/AOMediaCodec/libavif/blob/c673f2e884e635ff8e9b2c2951a2dddd2a00ffc3/src/write.c#L50